src/data/list/zip.lean

```diff
@@ -242,4 +264,25 @@ begin
         simp * at *, }, }, },
 end
 
+@[simp] lemma map_uncurry_zip_eq_zip_with
+  (f : α → β → γ) (l : list α) (l' : list β) :
+  map (function.uncurry f) (l.zip l') = zip_with f l l' :=
+begin
+  induction l with hd tl hl generalizing l',
+  { simp },
+  { cases l' with hd' tl',
+    { simp },
+    { simp [hl] } }
+end
+
+@[simp] lemma sum_zip_with_distrib_left (f : α → β → ℕ) (n : ℕ) (l : list α) (l' : list β) :
```

Comment:
This can either be generalized to work for comm_semiring (or whatever is appropriate) or just proved as a side hypothesis in your of_digits proof. The reason to do the latter might be that otherwise, adding the other mul lemma and the other two add lemmas would seem necessary to be complete.

Reply:
Indeed, maybe, adding the other mul lemma and using `@to_additive` can make this work?

Reply:
It appears to me that `semiring` is enough for this, I will do the change.